src/MagnumPlugins/BasisImageConverter/Test/BasisImageConverterTest.cpp

```diff
@@ -878,22 +918,107 @@ void BasisImageConverterTest::convert2DMipmaps() {
+           the image data properly */
+        level.imageWithSkip = copyImageWithSkip<Vector4>(ImageView2D(*originalImage), {7, 8});
+    }
+
+    /* Revert back to loading the actual format. BasisImporter can't transcode
+       to 32-bit float, which is fine since the test files are half float as
+       well. */
+    imageImporter->configuration().setValue("forceChannelType", "");
+
+    Containers::Pointer<AbstractImageConverter> converter = _converterManager.instantiate("BasisImageConverter");
+
+    /* HDR is encoded using UASTC HDR, but shouldn't require uastc to be
+       explicitly enabled */
+    CORRADE_COMPARE(converter->configuration().value<bool>("uastc"), false);
+
+    Containers::Optional<Containers::Array<char>> compressedData = converter->convertToData({*levels[0].imageWithSkip, *levels[1].imageWithSkip, *levels[2].imageWithSkip});
+    CORRADE_VERIFY(compressedData);
+
+    if(_manager.loadState("BasisImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("BasisImporter plugin not found, cannot test");
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporterRGBA16F");
+    CORRADE_VERIFY(importer->openData(*compressedData));
+    CORRADE_COMPARE(importer->image2DCount(), 1);
+    CORRADE_COMPARE(importer->image2DLevelCount(0), Containers::arraySize(levels));
+
+    for(std::size_t i = 0; i != Containers::arraySize(levels); ++i) {
+        CORRADE_ITERATION("level" << i);
+        auto result = importer->image2D(0, i);
+        CORRADE_VERIFY(result);
+
+        /* mutablePixels() so we can use the convenient slice syntax. Const
+           overloads of a()/rgb() return by value which makes it not work. */
+        const auto pixels = result->mutablePixels<Color4h>();
+
+        /* For HDR images alpha is always transcoded to 1. Can't use a
+           StridedArrayView with broadcasted size since CompareImage only
+           accepts StridedArrayView for the actual image, not the expected. */
+        using namespace Math::Literals;
+        Containers::Array<Half> ones{DirectInit, size_t(result->size().product()), 1.0_h};
+
+        CORRADE_COMPARE_WITH(pixels.slice(&Color4h::a),
+            (ImageView2D{PixelStorage{}.setAlignment(1), PixelFormat::R16F, result->size(), ones}),
+            /* No errors, always exactly 1.0 */
+            (DebugTools::CompareImage{0.0f, 0.0f}));
+
+        /* Drop alpha channel to only test RGB. Not using CompareImageToFile to
+           avoid overwriting the 4-channel expected .exr with a 3-channel
+           actual image when --save-diagnostic is specified. */
+        imageImporter->configuration().setValue("a", "");
+        
+        CORRADE_VERIFY(imageImporter->openFile(Utility::Path::join(BASISIMPORTER_TEST_DIR, levels[i].file)));
+        const auto expected = imageImporter->image2D(0);
+        CORRADE_VERIFY(expected);
+
+        imageImporter->configuration().setValue("a", "A");
```

Comment:
Instead of resetting the config values back, which makes it a bit uncertain what the importer is actually doing at given point, I'd create two instances, one for importing RGBA32F and one for RGB16F.

The config, if set on a plugin instance and not on the manager, is local to the instance, so they won't clash.